cmd/scollector/collectors/rabbitmq.go

```diff
@@ -0,0 +1,440 @@
+	Add(&md, p+"connections", o.ObjectTotals.Connections, ts, metadata.Gauge, metadata.Connection, DescRmqObjectTotalsConnections)
+	Add(&md, p+"consumers", o.ObjectTotals.Consumers, ts, metadata.Gauge, metadata.Consumer, DescRmqObjectTotalsConsumers)
+	Add(&md, p+"exchanges", o.ObjectTotals.Exchanges, ts, metadata.Gauge, metadata.Exchange, DescRmqObjectTotalsExchanges)
+	Add(&md, p+"queues", o.ObjectTotals.Queues, ts, metadata.Gauge, metadata.Queue, DescRmqObjectTotalsQueues)
+	msgStats := rabbitmqMessageStats(p, ts, o.MessageStats)
+	md = append(md, msgStats...)
+	return md, nil
+}
+
+func rabbitmqQueues(s string) (opentsdb.MultiDataPoint, error) {
+	p := rmqPrefix + "queue."
+	res, err := http.Get(s + "/api/queues")
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	content, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	var qs []rmqQueue
+	if err := json.NewDecoder(bytes.NewReader(content[:])).Decode(&qs); err != nil {
+		return nil, err
+	}
+	var md opentsdb.MultiDataPoint
+	for _, q := range qs {
+		host := strings.Split(q.Node, "@")[1]
+		ts := opentsdb.TagSet{"host": host, "queue": q.Name, "vhost": q.Vhost}
+		Add(&md, p+"consumers", q.Consumers, ts, metadata.Gauge, metadata.Consumer, DescRmqConsumers)
+		Add(&md, p+"memory", q.Memory, ts, metadata.Gauge, metadata.Bytes, DescRmqMemory)
+		Add(&md, p+"message_bytes_total", q.MessageBytes, ts, metadata.Gauge, metadata.Bytes, DescRmqMessageBytes)
+		Add(&md, p+"message_bytes_persistent", q.MessageBytesPersistent, ts, metadata.Gauge, metadata.Bytes, DescRmqMessageBytesPersistent)
+		Add(&md, p+"message_bytes_transient", q.MessageBytesRAM, ts, metadata.Gauge, metadata.Bytes, DescRmqMessageBytesRAM)
+		Add(&md, p+"message_bytes_ready", q.MessageBytesReady, ts, metadata.Gauge, metadata.Bytes, DescRmqMessageBytesReady)
+		Add(&md, p+"message_bytes_unack", q.MessageBytesUnacknowledged, ts, metadata.Gauge, metadata.Bytes, DescRmqMessageBytesUnacknowledged)
+		Add(&md, p+"messages_queue_depth", q.Messages, ts, metadata.Gauge, metadata.Message, DescRmqMessages)
+		Add(&md, p+"messages_persistent", q.MessagesPersistent, ts, metadata.Gauge, metadata.Message, DescRmqMessagesPersistent)
+		Add(&md, p+"messages_transient", q.MessagesRAM, ts, metadata.Gauge, metadata.Message, DescRmqMessagesRAM)
+		Add(&md, p+"messages_ready_total", q.MessagesReady, ts, metadata.Gauge, metadata.Message, DescRmqMessagesReady)
+		Add(&md, p+"messages_ready_transient", q.MessagesReadyRAM, ts, metadata.Gauge, metadata.Message, DescRmqMessagesReadyRAM)
+		Add(&md, p+"messages_unack_total", q.MessagesUnacknowledged, ts, metadata.Gauge, metadata.Message, DescRmqMessagesUnacknowledged)
+		Add(&md, p+"messages_unack_transient", q.MessagesUnacknowledgedRAM, ts, metadata.Gauge, metadata.Message, DescRmqMessagesUnacknowledgedRAM)
+		if sn, ok := q.SlaveNodes.([]interface{}); ok {
```

Comment:
I need some clarification.

Code works
```
2015/08/13 13:41:44 info: queue.go:66: {"metric":"rabbitmq.queue.slave_nodes","timestamp":1439462503,"value":2,"tags":{"host":"rabbit1","queue":"hello","vhost":"/"}}
2015/08/13 13:41:44 info: queue.go:66: {"metric":"rabbitmq.queue.sync_slave_nodes","timestamp":1439462503,"value":2,"tags":{"host":"rabbit1","queue":"hello","vhost":"/"}}
```
or do you mean like do we need those metrics at all? TBH I don't know that _yet_ because we are only planning to use mirror queues, but I'm fairly confident that we could use these.